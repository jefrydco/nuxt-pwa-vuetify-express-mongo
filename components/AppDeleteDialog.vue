<template>
  <v-dialog :value="value" width="500" @input="$emit('input', $emit)">
    <slot slot="activator" name="activator" />
    <v-card>
      <v-card-text>
        <v-container fluid="" grid-list-xl="">
          <v-layout row="" wrap="">
            <v-flex xs12="">
              Are you sure you want to permanently delete "<strong>{{
                name
              }}</strong
              >"?
            </v-flex>
          </v-layout>
        </v-container>
      </v-card-text>
      <v-divider />
      <v-card-actions>
        <v-spacer />
        <v-btn
          class="v-btn__custom--cancel-dialog"
          outline=""
          color="grey"
          @click="$emit('cancel')"
        >
          Cancel
        </v-btn>
        <v-btn
          class="v-btn__custom--delete-dialog"
          color="error"
          @click="$emit('delete')"
        >
          Delete
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  props: {
    value: {
      type: Boolean,
      required: true
    },
    name: {
      validator: prop => typeof prop === "string" || prop === null,
      required: true
    }
  }
};
</script>
