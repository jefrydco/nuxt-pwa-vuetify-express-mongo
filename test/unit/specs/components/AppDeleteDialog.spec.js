import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import VeeValidate from "vee-validate";

import AppDeleteDialog from "~/components/AppDeleteDialog";

// https://github.com/vuetifyjs/vuetify/issues/4068#issuecomment-446988490
const logError = console.error;
console.error = (...args) => {
  if (
    args[0].includes("[Vuetify]") &&
    args[0].includes("https://github.com/vuetifyjs/vuetify/issues/4068")
  )
    return;
  logError(...args);
};

const localVue = createLocalVue();
localVue.use(Vuetify);
localVue.use(VeeValidate);

describe("AppDeleteDialog component", () => {
  it("renders to match snapshot", () => {
    const wrapper = shallowMount(AppDeleteDialog, {
      localVue,
      propsData: {
        value: true,
        name: "Stub"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
