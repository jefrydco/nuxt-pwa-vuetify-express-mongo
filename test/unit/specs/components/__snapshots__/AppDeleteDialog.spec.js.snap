// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDeleteDialog component renders to match snapshot 1`] = `
<v-dialog-stub contentclass="" value="true" maxwidth="none" origin="center center" width="500" transition="dialog-transition"><template></template>
  <v-card-stub tag="div">
    <v-card-text-stub>
      <v-container-stub tag="div" fluid="" grid-list-xl="">
        <v-layout-stub tag="div" row="" wrap="">
          <v-flex-stub tag="div" xs12="">
            Are you sure you want to permanently delete "<strong>Stub</strong>"?
          </v-flex-stub>
        </v-layout-stub>
      </v-container-stub>
    </v-card-text-stub>
    <v-divider-stub></v-divider-stub>
    <v-card-actions-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-btn-stub color="grey" activeclass="v-btn--active" tag="button" outline="true" type="button" class="v-btn__custom--cancel-dialog">
        Cancel
      </v-btn-stub>
      <v-btn-stub color="error" activeclass="v-btn--active" tag="button" type="button" class="v-btn__custom--delete-dialog">
        Delete
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
